### PR TITLE
Modification for texture PR

### PR DESCRIPTION
**Pull request by JeremyAdamHart**
This implements the feedback I gave on: https://github.com/lakinwecker/poggles/pull/26

In addition, it includes changes that Lakin made to get rid of lint errors and warnings that are currently on upstream/main, as well as a couple small reorganizations that I found while making these changes

### CODE REVIEW

File: source/poggles/texture.cpp [hunk 1]
```diff
@@ -40,86 +13,100 @@ struct overload : Ts...
...
+
+auto poggles::loadFromData(poggles::texture& tex,
+                           std::variant<float*, uint8_t*> data,
+                           int width,
+                           int height,
+                           int channels,
+                           std::optional<GLenum> target,
+                           GLint level) -> bool
+{
+  glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
```
Comment (ebiederstadt) [inline source/poggles/texture.cpp]: Should this be `target` (if target is non-empty?)

**Review by ebiederstadt — CHANGES_REQUESTED** [review]
Thanks for taking the time to make these changes! Looks really great, I just have one small question :)

File: source/poggles/texture.cpp (hunk 1 above)
Comment (JeremyAdamHart) [reply source/poggles/texture.cpp]: Good catch, it's fixed now

**Review by ebiederstadt — APPROVED** [review]
LGTM!

**Pull request merged**